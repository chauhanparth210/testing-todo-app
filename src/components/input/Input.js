import React, { useState } from "react";

export const Input = ({ addTodo }) => {
  const [input, setInput] = useState("");

  const submitHandler = (e) => {
    e.preventDefault();
    addTodo(input);
    setInput("");
  };

  return (
    <form className="input-form" onSubmit={submitHandler}>
      <input
        type="text"
        value={input}
        onChange={(e) => setInput(e.target.value)}
      />
      <button type="submit">add todo</button>
    </form>
  );
};
