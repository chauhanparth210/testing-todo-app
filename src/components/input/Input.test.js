import React from "react";
import { shallow } from "enzyme";
import TestRenderer from 'react-test-renderer';
import { Input } from "./Input";

describe("Input Component testing", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Input debug />);
    expect(wrapper.exists()).toBeTruthy();
  });

  it("should render Input component properly", () => {
    const testRenderer = TestRenderer.create(<Input />);
    expect(testRenderer).toMatchSnapshot();
  });

  // it("change input values", () => {
  //   let inputElement = wrapper.find("input[type='text']");

  //   // initially the value should be empty
  //   expect(inputElement.props().value).toBe("");
  //   inputElement.simulate("change", { target: { value: "hello" } });

  //   // need to query again becuse it won't dynamically update the input value
  //   inputElement = wrapper.find("input[type='text']");
  //   expect(inputElement.props().value).toBe("hello");
  // });

  // it("adding a single todo item", () => {
  //   const addTodoMockFn = jest.fn();
  //   wrapper = shallow(<Input addTodo={addTodoMockFn} />);

  //   let inputElement = wrapper.find("input[type='text']");
  //   const formElement = wrapper.find(".input-form");

  //   // initially addTodoMockFn() should not be called and value should be empty
  //   expect(addTodoMockFn).not.toBeCalled();
  //   expect(inputElement.props().value).toBe("");

  //   inputElement.simulate("change", { target: { value: "hello" } });

  //   inputElement = wrapper.find("input[type='text']");
  //   expect(inputElement.props().value).toBe("hello");

  //   // press `Enter` or click on `add todo` button
  //   formElement.simulate("submit", {
  //     preventDefault: () => {},
  //   });

  //   inputElement = wrapper.find("input[type='text']");

  //   expect(inputElement.props().value).toBe("");

  //   jest.useFakeTimers();

  //   setTimeout(() => {
  //     expect(addTodoMockFn).toBeCalled();
  //     expect(addTodoMockFn).toHaveBeenCalledTimes(1);
  //   }, 1000);
  // });
});
