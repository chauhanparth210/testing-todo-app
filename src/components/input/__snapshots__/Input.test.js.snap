// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component testing should render Input component properly 1`] = `ShallowWrapper {}`;
